### PR TITLE
Dynamic loading libkvmi

### PR DESCRIPTION
**Pull request by Wenzel**
This PR loads `libkvmi.so` dynamically at runtime instead of requiring to link the binary with this library :tada: 

Note: The final binary still links with libkvmi, because I think we need to modify `kvmi-sys`crate also.

This PR wouldn't have been possible without @kmdouglass 's article, since I also struggled with `Symbol`'s lifetimes for a couple of hours until I have up and found his blog: :ok_hand: 
https://kmdouglass.github.io/posts/a-simple-plugin-interface-for-the-rust-ffi/

cc @rageagainsthepc

### CODE REVIEW

**Comment by Wenzel** [comment]
Also cc @arnabcs17b006

**Comment by Wenzel** [comment]
helps fix https://github.com/Wenzel/libmicrovmi/issues/68

File: src/lib.rs [hunk 1]
```diff
@@ -72,7 +76,7 @@ unsafe extern "C" fn new_guest_cb(
     _uuid: *mut [c_uchar; 16usize],
     cb_ctx: *mut c_void,
 ) -> c_int {
-    debug!("KVMi new guest");
+    info!("KVMi new guest");
```
Comment (rageagainsthepc) [inline src/lib.rs]: Why did you change the log level? This feels like something that should be debug level for me.

**Review by rageagainsthepc — APPROVED** [review]
lgtm

File: src/lib.rs (hunk 1 above)
Comment (Wenzel) [reply src/lib.rs]: I think both of these callbacks are important even in INFO logs because they tell whether the socket connexion has been established or not.

But thinking about it, we already log if the connexion is successful later on, so I can switch them back to debug.
